nix/checks/default.nix

```diff
@@ -8,9 +9,34 @@ let
     genAttrs
     ;
 
+  inherit (lib.fileset)
+    toSource
+    unions
+    ;
+
   inherit (lib.sources)
     cleanSource
     ;
+
+  # sources
+
+  # Used for derivations where facts is the primary directory.
+  factsSrc = toSource {
```

Comment:
Is the thought here that eventually we might have more (or less) required for the `factSrc`? Since at the moment they are the same dirs between `factsSrc` and `switchConfigurationSrc`

Reply:
It was just a coincidence that they share the same source. When I started with `factSrc`, I didn't realize it needed more that the `facts` directory because the scripts reach up and over into the `switch-configuration` directory for files. And then the same happened when I made `switchConfigurationSrc`. I started with it only having the `switch-configuration` directory but then the scripts failed because they reach into the `facts` directory.

I moved forward with keeping them separate because I wasn't sure if or how files would be added in new directories or consolidated into fewer. I thought about writing a fileset source in situ for each derivation. I also though about consolidating `factSrc` and `switchConfigurationSrc` into a single source. Opted to leave it as is for us to discuss.

Ideally, some time in the future, it would be ideal that these scripts don't reach across parent directories like they do but that can be tackled another day.

Reply:
> Hmm let me look at why this is. iirc theres some odd behavior with gomplate that Ive run into before. Id be nice to get all of these tests fixed while its top of mind.

Yeah I hacked at it for a couple hours and kept running into issues with gomplate. FYI if you want to hack on this, switch to using `mkDerivation` like I did for `perl-switches`. It sets write permissions on the source that you don't get with `runCommand`.